NickvisionMoney.GNOME/Views/AccountView.cs

```diff
@@ -329,11 +334,19 @@ private IGroupRowControl CreateGroupRow(Group group, int? index)
         row.FilterChanged += UpdateGroupFilter;
         if (index != null)
         {
-            _groupsList.Insert(row, index.Value);
+            g_main_context_invoke(IntPtr.Zero, (x) =>
+            {
+                _groupsList.Insert(row, index.Value);
+                return false;
+            }, IntPtr.Zero);
         }
         else
         {
-            _groupsList.Append(row);
+            g_main_context_invoke(IntPtr.Zero, (x) =>
+            {
+                _groupsList.Append(row);
+                return false;
+            }, IntPtr.Zero);
```

Comment:
I'm afraid this and all similar blocks can cause the infamous "A callback was made on a garbage collected delegate...", this is what I got when tried to use lambda in TransactionRow instead of separate UpdateRowSync. *\*sigh\** We'll probably need an additional ton of marshalling to do something like this:
```
g_main_context_invoke(IntPtr.Zero, GroupListAppend, row.Handle);
...
private bool GroupListAppend(nint rowHandle)
{
    gtk_listbox_append(_groupList.Handle, rowHandle);
    return false;
}
```

Or do you have a better idea?

Reply:
I found out about GCHandle, it can be used to convert between a pointer and a managed object:
```
var gch = GCHandle.Alloc(row);
g_main_context_invoke(IntPtr.Zero, GroupListAppend, (IntPtr)gch);
...
private bool GroupListAppend(nint rowHandle)
{
    var gch = (GCHandle)rowHandle;
    var row = (GroupRow)gch.Target;
    _groupsList.Append(row);
    gch.Free();
    return false;
}
```
It works. So maybe we could use it to avoid marshalling of all needed GTK methods?

Reply:
I wouldn't want to get too deep into manual memory management...it's ok for saving the callbacks manually for now if needed until gir.core has proper events for them.